File: FootballClubBackend/Model/Statistics/MatchStatistics.cs
using System.Reflection;
using System.Runtime.InteropServices;

namespace FootballClubBackend.Model.Statistics
{
    public class MatchStatistics
    {
        public GeneralMatchStatistics GeneralMatchStatistics { get; set; }
        public AttackingMatchStatistics AttackingMatchStatistics { get; set; }
        public PassingMatchStatistics PassingMatchStatistics { get; set; }
        public DefendingMatchStatistics DefendingMatchStatistics { get; set; }
    }

    public class GeneralMatchStatistics
    {
        public float Possession { get; set; }
        public float DuelSuccessRate { get; set; }
        public float AerialDuelSuccessRate { get; set; }
        public int Interceptions { get; set; }
        public int Offsides { get; set; }
        public int Corners { get; set; }
        public int Passes { get; set; }
    }

    public class AttackingMatchStatistics
    {
        public int Goals { get; set; }
        public int Shots { get; set; }
        public int ShotsOnTarget { get; set; }
        public int BlockedShots { get; set; }
        public int ShotsOutsideTheBox { get; set; }
        public int ShotsInsideTheBox { get; set; }
        public float ShootingAccuracy { get; set; }
    }

    public class PassingMatchStatistics
    {
        public int LongPasses { get; set; }
        public float PassingAccuracy { get; set; }
        public float PassingAccuracyInOpponentsHalf { get; set; }
        public int Crosses { get; set; }
        public float CrossingAccuracy { get; set; }
    }

    public class DefendingMatchStatistics
    {
        public int Tackles { get; set; }
        public float TackleSuccessRate { get; set; }
        public int Clearances { get; set; }
        public int YellowCards { get; set; }
        public int RedCards { get; set; }
        public int Fouls { get; set; }
    }
}

File: FootballClubFrontend/src/app/model/match-statistics.ts
export class MatchStatisticsDto{
     statistics !: MatchStatistics
     opponentStatistics !: MatchStatistics
}

export class MatchStatistics
{
     generalMatchStatistics !: GeneralMatchStatistics
     attackingMatchStatistics !: AttackingMatchStatistics 
     passingMatchStatistics !: PassingMatchStatistics 
     defendingMatchStatistics !: DefendingMatchStatistics 
}

export class GeneralMatchStatistics
{
     possession : number = 0
     duelSuccessRate : number = 0
     aerialDuelSuccessRate : number = 0
     interceptions : number = 0
     offsides : number = 0
     corners : number = 0
     passes : number = 0
}

export class AttackingMatchStatistics
{
     goals : number = 0
     shots : number = 0
     shotsOnTarget : number = 0
     blockedShots : number = 0
     shotsOutsideTheBox : number = 0
     shotsInsideTheBox : number = 0
     shootingAccuracy : number = 0
}

export class PassingMatchStatistics
{
     longPasses : number = 0
     passingAccuracy : number = 0
     passingAccuracyInOpponentsHalf : number = 0
     crosses : number = 0
     crossingAccuracy : number = 0
}

export class DefendingMatchStatistics
{
     tackles : number = 0
     tackleSuccessRate : number = 0
     clearances : number = 0
     yellowCards : number = 0
     redCards : number = 0
     fouls : number = 0
}